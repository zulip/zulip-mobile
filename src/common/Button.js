import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  ActivityIndicator,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'stretch',
    flex: 1,
    height: 44,
    alignSelf: 'stretch',
    justifyContent: 'center',
    borderRadius: 5,
    marginBottom: 10,
  },
  primaryContainer: {
    backgroundColor: '#22693F',
  },
  secondaryContainer: {
    borderColor: '#22693F',
    borderWidth: 2,
  },
  touchTarget: {
    flex: 1,
    alignSelf: 'stretch',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: '#FFFFFF',
  },
  primarytext: {
    color: 'white',
  },
  secondaryText: {
    color: '#22693F',
  },
});

export default class Button extends React.PureComponent {

  props: {
    customStyles: Object,
    progress: boolean,
    text: string,
    secondary: boolean,
    onPress: () => void,
  }

  render() {
    const { customStyles, text, secondary, progress, onPress } = this.props;
    const containerStyle = [
      customStyles,
      styles.container,
      secondary ? styles.secondaryContainer : styles.primaryContainer,
    ];
    const textStyle = [
      styles.text,
      secondary ? styles.secondaryText : styles.primaryText,
    ];

    return progress ?
      <View style={containerStyle}>
        <ActivityIndicator />
      </View> :
      <View style={containerStyle}>
        <TouchableHighlight
          style={styles.touchTarget}
          underlayColor="rgba(34, 105, 63, 0.5)"
          onPress={onPress}
        >
          <Text style={textStyle}>
            {text}
          </Text>
        </TouchableHighlight>
      </View>;
  }
}
