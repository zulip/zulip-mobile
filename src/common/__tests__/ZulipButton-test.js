import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

it('renders correctly', () => {
  const tree = renderer.create(
    <Button />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
