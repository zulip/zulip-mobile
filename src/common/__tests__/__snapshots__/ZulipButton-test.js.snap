exports[`test renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderRadius": 5,
        "height": 44,
        "justifyContent": "center",
        "marginBottom": 10
      },
      Object {
        "backgroundColor": "#22693F"
      }
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flex": 1,
          "justifyContent": "center"
        }
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF"
          },
          undefined
        ]
      } />
  </View>
</View>
`;
